### PR TITLE
docs:  iOS Runtime Types

### PR DESCRIPTION
**Pull request by Ombuweb**
- I need help with the examples for these docs.

### CODE REVIEW

File: content/guide/ios-runtime-types.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+---
+title: iOS Runtime Types
+---
+
+NativeScript provides the `interop` module for working with native C types, pointers, pointer arithmetic and memory.
+
+
+<!-- TODO: Add examples -->
```
Comment (SeanKelly369) [inline content/guide/ios-runtime-types.md:8]: Don't forget the TODO

File: content/guide/ios-runtime-types.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+## Using interop
+
+### Creating a pointer
+
+The `Pointer` type allows you to represent a void*.
+
+#### Using the constructor
+
+To create a new pointer with the given offset, use the construcotr:
```
Comment (SeanKelly369) [inline content/guide/ios-runtime-types.md]:
```suggestion
To create a new pointer with the given offset, use the constructor:
```

File: content/guide/ios-runtime-types.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+```ts
+   const interop1 = new interop.Pointer(34)
+```
+
+#### Creating a pointer by adding an offeset from an existing pointer
```
Comment (SeanKelly369) [inline content/guide/ios-runtime-types.md]:
```suggestion
#### Creating a pointer by adding an offset from an existing pointer
```

File: content/guide/ios-runtime-types.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+ ```ts
+const interop2 = interop1.add(4)
+ ```
+
+#### Creating a pointer by removing an offeset to an existing pointer
```
Comment (SeanKelly369) [inline content/guide/ios-runtime-types.md]:
```suggestion
#### Creating a pointer by removing an offset to an existing pointer
```

File: content/guide/ios-runtime-types.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+### handleof
+
+```ts
+
+interop.handleof(instance: any): Pointer
+
+```
+
+From a JavaScript object gets a pointer to the backing native object. The `instance` can be a class constructor (of Objective-C interface), an instance (wrapper of Objective-C interface), struct instance, reference, protocol, function (for c function) or block.
```
Comment (SeanKelly369) [inline content/guide/ios-runtime-types.md]:
```suggestion
A JavaScript object can obtain a reference to the underlying native object. The instance can represent a class constructor (for an Objective-C interface), an instance (wrapper of an Objective-C interface), a struct instance, a reference, a protocol, a function (for C functions), or a block.
```

File: content/guide/ios-runtime-types.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+```
+
+Wraps an NSData instance in an ArrayBuffer.
+
+---
+
+
+### new Reference(value)
+
+Creates a new reference around a JavaScript `value`. The native representation of the type will be determined the first time the Reference is used in operation involving marshalling.
```
Comment (SeanKelly369) [inline content/guide/ios-runtime-types.md]:
```suggestion
Creates a new reference around a JavaScript `value`. The native representation of the type will be determined during the first time the Reference is used in operation involving marshalling.
```

File: content/guide/ios-runtime-types.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+---
+
+### new Reference(type: Type\<T\>, data: Pointer)
+
+Creates a reference from to the pointer with a given type.
```
Comment (SeanKelly369) [inline content/guide/ios-runtime-types.md]:
```suggestion
Creates a reference from the pointer with a given type.
```

File: content/guide/ios-runtime-types.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+
+### new Reference(type: Type\<T\>, value: any)
+
+Creates a new reference around a `value` of a certain `type`.
+
+---
+
+### new FunctionReference(func: T)
+
+Creates a function reference that can be marshalled as a native function pointer. The JavaScript reference must be held alive as long as the native code needs the function.
```
Comment (SeanKelly369) [inline content/guide/ios-runtime-types.md]:
```suggestion
Creates a function reference that can be marshalled as a native function pointer. The JavaScript reference must be allocated in memory as long as the native code needs the function.
```

File: content/guide/ios-runtime-types.md [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+someStructTypeIntastance.equals(left: T, right: T): boolean
+
+```
+
+Checks two structs for equality.
+
+---
+
+### new NSErrorWrapper (error: NSError)
+<!-- TODO: Add description -->
```
Comment (SeanKelly369) [inline content/guide/ios-runtime-types.md]: Reminder

File: content/guide/ios-runtime-types.md (hunk 5 above)
Comment (NathanWalker) [reply content/guide/ios-runtime-types.md]: really wonderful touches on this doc @SeanKelly369 thank you!

**Pull request merged**